common/ledger/blockledger/fileledger/factory.go

```diff
@@ -90,8 +101,28 @@ func New(directory string, metricsProvider metrics.Provider) (blockledger.Factor
 	if err != nil {
 		return nil, err
 	}
+
+	fileRepo, err := filerepo.New(directory, "remove")
+	if err != nil {
+		return nil, err
+	}
+
+	files, err := fileRepo.List()
+	if len(files) != 0 {
+		for _, fileName := range files {
+			channelName := fileRepo.FileToBaseName(fileName)
+			err := fileRepo.Remove(channelName)
```

Comment:
You have to actually complete removing the channel here, with the equivalent of
```go
err := f.blkstorageProvider.Drop(channelID)
if err != nil {
   return err
}

delete(f.ledgers, channelID)
```
and only then remove the filerepo marker. The fact that we have a marker here means the `Remove` did not complete before the server crashed.

Reply:
🧹